src/main/java/net/mcreator/ui/modgui/ModElementGUI.java

```diff
@@ -529,10 +529,6 @@ private void finishModCreation(boolean closeTab) {
 		MCREvent.event(new ModElementGUIEvent.WhenSaving(mcreator, tabIn, this, !closeTab));
 		GE element = getElementFromGUI();
 
-		// if new element, specify the folder of the mod element
-		if (!editingMode)
```

Comment:
NewModElementDialog is not the only way how a new mod element can be created and thus this call is still needed for such cases

Reply:
How do we know then that the folder was set earlier? Without removing this, parent folder setting from the modified dialog will always be ignored.

Reply:
The point is, if you don't know how to do something, it does not change the fact that you _can't just silently remove it and ignore the consequences.

One option I see is a new field in ModElementGUI shouldUpdateFolderOnCreating with default value to true and method to set it to false and calling this method when appropriate

Reply:
Is current solution fine (`FolderElement targetFolder`)?

Reply:
Yes